web3/wallets.js

```diff
@@ -505,6 +506,51 @@ exports.getAllWallets = async (req, res) => {
     } else if (Object.keys(res).length !== 0)
         return res.status(401).end('Account not found')
 }
+const getNetworkByToken = async (idCrypto) => {
+
+    try{
+        if (
+            cache.get('networks') &&
+            Date.now() - new Date(cache.get('networks')?.date).getTime() < 604800000
+        ) {
+            return cache.get('networks').data
+        }else{
+    
+        const options = {
+            method: 'GET',
+            url: process.env.CMC_CRYPTO_DETAILS,
+            params: {
+                id: idCrypto
+            },
+            headers: {
+                'X-CMC_PRO_API_KEY': process.env.CMCAPIKEY,
+            },
+           
+        }
+       
+        var results 
+        try {
+            results = await Promise.all([ rp.request(options)])
```

Comment:
Can you clarify why Promise.all is used here? As I understand, Promise.all is generally used when dealing with multiple promises. However, it seems like we're only making a single request in this instance. If we're only making a single request, we can call this request directly without using Promise.all.